src/main/java/book/store/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+import book.store.onlinebookstore.dto.order.OrderDto;
+import book.store.onlinebookstore.dto.order.UpdateOrderRequestDto;
+import book.store.onlinebookstore.dto.order.item.OrderItemDto;
+import book.store.onlinebookstore.exception.EntityNotFoundException;
+import book.store.onlinebookstore.mapper.OrderItemMapper;
+import book.store.onlinebookstore.mapper.OrderMapper;
+import book.store.onlinebookstore.model.CartItem;
+import book.store.onlinebookstore.model.Order;
+import book.store.onlinebookstore.model.OrderItem;
+import book.store.onlinebookstore.model.ShoppingCart;
+import book.store.onlinebookstore.repository.order.OrderRepository;
+import book.store.onlinebookstore.repository.order.item.OrderItemRepository;
+import book.store.onlinebookstore.repository.shoppingcart.ShoppingCartRepository;
+import book.store.onlinebookstore.repository.user.UserRepository;
+import book.store.onlinebookstore.service.OrderService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final UserRepository userRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderDto placeOrder(Long userId, CreateOrderRequestDto requestDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException("Your shopping cart is empty"));
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+
+        Order newOrder = initNewOrder(requestDto, userId);
+        orderRepository.save(newOrder);
+        BigDecimal total = BigDecimal.ZERO;
+        Set<OrderItem> orderItems = cartItems.stream()
+                .map(cartItem -> getOrderItem(cartItem, newOrder, total))
+                .collect(Collectors.toSet());
+        newOrder.setOrderItems(orderItems);
+
+        shoppingCartRepository.deleteById(userId);
+        return orderMapper.toDto(newOrder);
+    }
+
+    @Override
+    public List<OrderDto> getOrders(Long userId, Pageable pageable) {
+        List<Order> allByUserId = orderRepository.findAllByUserId(userId, pageable);
+        return allByUserId.stream().map(orderMapper::toDto).toList();
+    }
+
+    @Override
+    @Transactional
+    public List<OrderItemDto> getOrderItemsByOrderId(Long userId, Long orderId) {
+        Order order = orderRepository.findByIdAndUserId(orderId, userId)

```

Comment:
Try another query. For example
List<OrderItem> orderItems = orderItemRepository.findAllByOrderId(orderId, pageable);
return orderItemMapper.toDtoList(orderItems);